src/extract/mod.rs

```diff
@@ -198,3 +201,90 @@ impl ExtractionResult {
                 .sum::<Cost>()
     }
 }
+
```

Comment:
Move this code to the new testing module as well.

Reply:
yes, likewise.